wolox-books/src/app/guards/auth.guard.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import {
+  CanActivate,
+  Router,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { UserService } from '../services/user.service';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+  constructor(private userService: UserService, private router: Router) {}
+
+  canActivate(): boolean {
+    const hasToken = this.userService.hasToken();
+    if (hasToken) {
+      return true;
+    } else {
+      this.router.navigate(['/login']);
+      return false;
+    }
```

Comment:
Mmm, maybe this is cleaner
```suggestion
    if (!hasToken) this.router.navigate(['/login']);
    return hasToken;
```